pywbem/_logging.py

```diff
@@ -14,371 +14,303 @@
-        """
-        Create the pywbem loggers defined by the input string in the following
-        syntax and place the logger definitions in the class level dictionary
-        in this class.
+Configure a single pywbem connection with standard Python logger methods::
 
-        Parameters:
+    import logging
+    logging.basicConfig(filename='example.log',level=logging.DEBUG)
+    conn = WBEMConnection(...)
+    conn.configure_all_loggers(detail_level='all', connection=conn)
 
-          input_str (:term:`string`): Specifies the logger definitions
-            as follows:
+Set up configuration to enable the api logger for all subsequent WBEMConnections
+using Python methods to configure the loggers::
 
-            ``log_specs`` := ``log_spec`` [, ``log_spec`` ]
+    logger = logging.getLogger('pywbem')
+    handler = logging.StreamHandler()
+    handler.setLevel(logging.DEBUG)
+    formatter = logging.Formatter('%(asctime)s - %(name)s - %(message)s')
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
 
-            ``log_spec`` := ``log_comp`` ['=' [ ``dest`` ][":"[ ``detail_level`` ]]]]
+    WBEMConnection.configure_api_logger(detail_level='summary', connection=True)
 
-            where:
+    # all of these connections do api call and response logging with
+    # summary information.
+    conn.WBEMConnection(...)
+    conn2.WBEMConnection(...)
+
+
+Configure the api logger for summary information output using only pywbem
+methods to configure the logger::
 
-            * ``log_comp``: Must be one of strings in the
-              :data:`~pywbem._logging.LOG_COMPONENTS` list.
+    WBEMConnection.configure_api_logger(dest='stderr',
+                                     detail_level='summary', connection=True)
 
-            * ``detail_level``: Must be one of strings in the
-              :data:`~pywbem._logging.LOG_DETAIL_LEVELS` list.
+    conn = WBEMConnection()
 
-            * ``dest``: Must be one of strings in the
-              :data:`~pywbem._logging.LOG_DESTINATIONS` list.
+Configure just a single WBEMConnection for output of summary information::
 
-          log_filename (:term:`string`)
-            Optional string that defines the filename for output of logs
-            if the dest type is `file`
+    conn = WBEMConnection()
+    conn.configure_api_logger(dest='file', log_filname='xxx.log',
+                              connection=conn)
 
-        Raises:
+Log everything to stderr and output all of the information using the
+pywbem method that configures the loggers from a string::
 
-          ValueError: Generated if the syntax of the input string is invalid
-            or any of the components is not one of allowed strings.
+    define_loggers_from_string('all=stderr':summary', connection=True)
+    conn = WBEMConnection()
 
-        Examples::
+or::
 
-            ops=stderr:min    # set cim operations logger
+    define_loggers_from_string('api=stderr':summary,http=summary',
```

Comment:
One single quote too much in the first argument.

Reply:
Done. Checked all of them